src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -1019,22 +1038,6 @@ std::pair<std::vector<String>, bool> ReplicatedMergeTreeSinkImpl<async_insert>::
     {
         zookeeper->setKeeper(storage.getZooKeeper());
```

Comment:
I'm not sure about moving this block to `lock and commit` stage. Probably, you could explain

Reply:
I moved it because the table being in read only only matters when we are going to add a new part/commit to ZK. If we only have checks left (for example in `resolve_duplicate_stage`) it doesn't make sense to not continue verifying because the table is stopped, as we can still return a proper response to the client.

A similar thing will happen for quorum for example (but it's not in this PR to keep in more concise).  It's ok to check for quorum if the write succeeded but the table was set in read only after that.